genesis/example/member.go

```diff
@@ -0,0 +1,163 @@
+/*
+ *    Copyright 2018 INS Ecosystem
+ *
+ *    Licensed under the Apache License, Version 2.0 (the "License");
+ *    you may not use this file except in compliance with the License.
+ *    You may obtain a copy of the License at
+ *
+ *        http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *    Unless required by applicable law or agreed to in writing, software
+ *    distributed under the License is distributed on an "AS IS" BASIS,
+ *    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *    See the License for the specific language governing permissions and
+ *    limitations under the License.
+ */
+
+package example
+
+import (
+	"fmt"
+
+	"github.com/insolar/insolar/genesis/model/domain"
+	"github.com/insolar/insolar/genesis/model/factory"
+	"github.com/insolar/insolar/genesis/model/object"
+	"github.com/insolar/insolar/genesis/model/resolver"
+)
+
+// MemberDomainName is a name for member domain.
+const MemberDomainName = "MemberDomain"
+
+// MemberDomainID is a string representation of class for MemberDomain interface implementations.
+const MemberDomainID = "MemberDomain"
+
+// MemberDomain is a contract that allows to add new members to system.
+type MemberDomain interface {
+	// Base domain implementation.
+	domain.Domain
+	// CreateMember is used to create new member as a child to domain storage.
+	CreateMember(factory.Factory) (string, error)
+	// GetMember returns member from its record in domain storage.
+	GetMember(string) (resolver.Proxy, error)
+}
+
+type memberDomain struct {
+	domain.BaseDomain
+}
+
+// newMemberDomain creates new instance of MemberDomain.
+func newMemberDomain(parent object.Parent) (*memberDomain, error) {
+	if parent == nil {
+		return nil, fmt.Errorf("parent must not be nil")
+	}
+
+	return &memberDomain{
+		BaseDomain: *domain.NewBaseDomain(parent, MemberDomainName),
+	}, nil
+}
+
+// GetClassID returns string representation of MemberDomain's class.
+func (md *memberDomain) GetClassID() string {
+	return MemberDomainID
+}
+
+// CreateMember creates new member as a child to domain storage.
+func (md *memberDomain) CreateMember(fc factory.Factory) (string, error) {
+	member, err := fc.Create(md)
+	if err != nil {
+		return "", err
+	}
+	if member == nil {
+		return "", fmt.Errorf("factory returns nil")
+	}
+
+	record, err := md.ChildStorage.Set(member)
+	if err != nil {
+		return "", err
+	}
+
+	return record, nil
+}
+
+// GetMember returns member from its record in domain storage.
+func (md *memberDomain) GetMember(record string) (resolver.Proxy, error) {
```

Comment:
Here you also need to rely on MemberProxy class declaration,  not Proxy

Reply:
Then Member class will be created, it will be done.